### PR TITLE
Grpc telemetry

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [x] Have you checked your code using ```cargo clippy``` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/segment/src/telemetry.rs [hunk 1]
```diff
@@ -151,7 +151,7 @@ impl TelemetryOperationAggregator {
             let from = if i < SLIDING_WINDOW_LEN {
                 0
             } else {
-                SLIDING_WINDOW_LEN - i
+                i - SLIDING_WINDOW_LEN
```
Comment (agourlay) [inline lib/segment/src/telemetry.rs:154]: do you have a test for this or you just noticed the issue by chance?

File: src/actix/actix_telemetry.rs [hunk 2]
```diff
@@ -1,19 +1,20 @@
+use std::future::{ready, Ready};
 use std::sync::Arc;
 
 use actix_web::dev::{Service, ServiceRequest, ServiceResponse, Transform};
 use actix_web::Error;
 use futures_util::future::LocalBoxFuture;
-use tokio::sync::Mutex;
+use parking_lot::Mutex;
```
Comment (agourlay) [inline src/actix/actix_telemetry.rs:7]: why is it necessary to use `parking_lot` here?

File: src/common/telemetry.rs [hunk 3]
```diff
@@ -31,6 +58,7 @@ pub struct TelemetryData {
     configs: ConfigsTelemetry,
     collections: Vec<CollectionTelemetry>,
     web: WebApiTelemetry,
+    grpc_calls_statsistics: TelemetryOperationStatistics,
```
Comment (agourlay) [inline src/common/telemetry.rs:61]:
```suggestion
    grpc_calls_statistics: TelemetryOperationStatistics,
```

File: lib/segment/src/telemetry.rs (hunk 1 above)
Comment (IvanPleshkov) [reply lib/segment/src/telemetry.rs:154]: It was my mistake from prev PR. Prev PR I tested with small amount of collected data and didn't catch this bug

File: src/actix/actix_telemetry.rs (hunk 2 above)
Comment (IvanPleshkov) [reply src/actix/actix_telemetry.rs:7]: Regarding this description of tokio mutex
https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use
I'm using parking lot mutex where possible (but not everywhere). In this actix middleware we have 2 mutex-covered structures: for the first mutex is unique per working thread. for the second it is used only on actix start.

File: src/common/telemetry.rs (hunk 3 above)
Comment (IvanPleshkov) [reply src/common/telemetry.rs]: Thanks! Fixed

**Review by agourlay — APPROVED** [review]
The Tower stuff is a bit over my head but I am fine with it as you tested it manually.

**Pull request merged**